components/style/element_flags.rs

```diff
@@ -0,0 +1,24 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+bitflags! {
+    /// Set of flags that are set on the parent element depending on whether a
+    /// child matches a selector.
+    ///
+    /// These setters, in the case of Servo, must be atomic, due to the parallel
+    /// traversal.
+    pub flags ElementFlags: u8 {
```

Comment:
Let's call these TraversalGeneratedElementFlags, and update the comment appropriately. Filename is fine.